src/submodule.c

```diff
@@ -832,6 +832,47 @@ int git_submodule_repo_init(
 	return error;
 }
 
+static int clone_return_origin(git_remote **out, git_repository *repo, const char *name, const char *url, void *payload)
+{
+	GIT_UNUSED(url);
+	GIT_UNUSED(payload);
+	return git_remote_lookup(out, repo, name);
+}
+
+static int clone_return_repo(git_repository **out, const char *path, int bare, void *payload)
+{
+	git_submodule *sm = payload;
+
+	GIT_UNUSED(path);
+	GIT_UNUSED(bare);
+	return git_submodule_open(out, sm);
+}
+
+int git_submodule_add_clone(git_repository **out, git_submodule *submodule)
```

Comment:
I'd be tempted to say that the "old" way already provides this, and shrug it off. I'm adding this API for the 95% of users that just want to `git submodule add` something and don't care for wacky remote configurations.

Reply:
This function only clones and doesn't do the add, right? Wouldn't `git_submodule_clone` be easier to understand in that case?

Reply:
I… hadn't anticipated it would make sense 😆. My thinking was that, being the step between `add_setup` and `add_finalize`, it made sense to have it under that namespace. If it so happens that it can be used to clone actual uncloned submodules, then why not, but that wasn't the plan (and I'm eyeing towards the unimplemented checkout actions 😉).

Reply:
So, uh. Can it be used to clone uncloned submodules right now? If so, I'd rename it. Otherwise it's fine as is, I guess.